app/src/main/java/com/example/kitchenflow/ui/main/MainViewModel.kt

```diff
@@ -1,45 +1,123 @@
 package com.example.kitchenflow.ui.main
 
+import android.annotation.SuppressLint
 import android.icu.util.Calendar
+import android.view.View
 import androidx.databinding.ObservableField
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import com.example.kitchenflow.data.entity.OrderModel
+import com.example.kitchenflow.data.entity.VenueOpeningHours
 import com.example.kitchenflow.data.repository.KitchenRepository
 import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
 import io.reactivex.rxjava3.schedulers.Schedulers
 import java.text.SimpleDateFormat
 import java.util.*
+import java.util.concurrent.TimeUnit
 
 class MainViewModel(private val repository: KitchenRepository) : ViewModel() {
     private val ordersData: MutableLiveData<List<OrderModel>> = MutableLiveData()
     val orders: LiveData<List<OrderModel>> = ordersData
-    val date: ObservableField<String> = ObservableField(
+    var date: MutableLiveData<Date> = MutableLiveData(
         getCurrentDate()
     )
-
-    private fun getCurrentDate() = SimpleDateFormat("EEE, MMMM dd", Locale.ENGLISH).format(
-        Calendar.getInstance().time
+    lateinit var venueOpeningHours: VenueOpeningHours
+    val dateAndTime: ObservableField<String> = ObservableField(
+        SimpleDateFormat("EEE, MMMM dd", Locale.ENGLISH).format(
+            date.value?.time
+        )
     )
 
+    private fun getCurrentDate() = Calendar.getInstance().time
+
     val currentOrders: ObservableField<String> = ObservableField("0")
 
     init {
+        getVenueOpeningHours()
         fetchOrders()
     }
 
+    private fun getVenueOpeningHours() {
+        repository.getVenueOpeningHours()
+            .observeOn(Schedulers.io())
+            .subscribeOn(AndroidSchedulers.mainThread())
+            .subscribe {
+                venueOpeningHours = it
+            }
+    }
+
+    @SuppressLint("SimpleDateFormat")
+    fun isRestaurantOpen(): Boolean {
+        var calendar = Calendar.getInstance()
+        date.value?.month?.let { calendar.set(Calendar.MONTH, it) }
+        val annualClosure =
+            venueOpeningHours.annualClosures.find {
+                it.date == "${
+                    SimpleDateFormat("MMMM").format(calendar.time).uppercase()
+                }/${date.value?.date}"
```

Comment:
why don't put calculation of into val outside .find
"${SimpleDateFormat("MMMM").format(calendar.time).uppercase()}/${date.value?.date}" 
it will create same calculation for each find iteration
And again working with date as string :neutral_face: